pkg/kubectl/cmd/util/factory.go

```diff
@@ -307,82 +307,33 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
-			if discoverDynamicAPIs {
```

Comment:
discoverDynamicAPIs is not used anymore. The comment above seems to be out of place now.